### PR TITLE
[WIP] Use CSS Custom Properties for all Sass variables

### PR DESCRIPTION
**Pull request by clhenrick**
- allow for using Sass variables in the `.svelte` components as CSS custom properties
- allow for accessing the Sass variables in JavaScript as CSS custom properties
- fixes the ESLint configuration (should have been done in #8 but was missed)
- adds Sass and CSS vars for Carbon spacing in rem units
- adds an npm script for test building the Sass as it's faster than waiting on Sapper to do it.

### CODE REVIEW

**Comment by clhenrick** [comment]
@mukhtyar in this PR we may also want to go through the existing Svelte components and replace the hard coded color variables in the `<style>` blocks with their equivalent CSS Custom properties. This will make them easier to track down in the future if we end up updating our Sass/CSS variables and/or Cal Adapt theme colors.

**Comment by clhenrick** [comment]
We should probably add variables for [spacing](https://www.carbondesignsystem.com/guidelines/spacing/overview) as well. It supposedly can be imported from the [@carbon/layout](https://github.com/carbon-design-system/carbon/tree/main/packages/layout/scss) package which should be installed with the carbon components library.

File: src/routes/tools/climate-extremes/Header.svelte [hunk 1]
```diff
@@ -23,7 +23,7 @@
     top: 0;
     border-bottom: 1px solid #dadee1;
     z-index: 3;
-    background: #9fa6ac;
+    background: #var(--gray-30);
```
Comment (mukhtyar) [inline src/routes/tools/climate-extremes/Header.svelte]: Do we need a `#` here?

File: src/routes/tools/annual-averages/Header.svelte [hunk 2]
```diff
@@ -23,7 +23,7 @@
     top: 0;
     border-bottom: 1px solid #dadee1;
     z-index: 3;
-    background: #9fa6ac;
+    background: #var(--gray-30);
```
Comment (mukhtyar) [inline src/routes/tools/annual-averages/Header.svelte]: Do we need a `#` here?

**Review by mukhtyar — APPROVED** [review]
Thanks for taking care of this!

File: src/routes/tools/climate-extremes/Header.svelte (hunk 1 above)
Comment (clhenrick) [reply src/routes/tools/climate-extremes/Header.svelte]: fixed.

File: src/routes/tools/annual-averages/Header.svelte (hunk 2 above)
Comment (clhenrick) [reply src/routes/tools/annual-averages/Header.svelte]: fixed.

**Comment by clhenrick** [comment]
@mukhtyar just FYI I also added Sass and CSS variables for the Carbon spacing units which are mostly in multiples of 4 and 8 pixels. We can use these in the future for things like padding, margin, grid-gap, etc. to help keep the layout consistent between pages and to not have to worry about doing pixel to rem calculations.

**Pull request merged**